tests/unit/src/test/scala/tests/codeactions/CompanionObjectSuite.scala

```diff
@@ -0,0 +1,471 @@
+       |""".stripMargin,
+    selectedActionIndex = 1
+  )
+
+  check(
+    "insert companion object of trait inside package",
+    """|package baz
+       |
+       |trait F<<>>oo {
+       |
+       |}
+       |
+       |class Bar {
+       |
+       |}
+       |""".stripMargin,
+    s"""|${ExtractRenameMember.title("trait", "Foo")}
+        |${CreateCompanionObjectCodeAction.companionObjectCreation}
+        |""".stripMargin,
+    """|package baz
+       |
+       |trait Foo {
+       |
+       |}
+       |
+       |object Foo {
+       |
+       |}
+       |
+       |class Bar {
+       |
+       |}
+       |""".stripMargin,
+    selectedActionIndex = 1
+  )
+
+  check(
+    "insert companion object of trait inside parent object",
+    """|object Baz {
+       |  trait F<<>>oo {
+       |
+       |  }
+       |
+       |  class Bar {
+       |
+       |  }
+       |}
+       |""".stripMargin,
+    s"""|${CreateCompanionObjectCodeAction.companionObjectCreation}
+        |""".stripMargin,
+    """|object Baz {
+       |  trait Foo {
+       |
+       |  }
+       |
+       |  object Foo {
+       |
+       |  }
+       |
+       |  class Bar {
+       |
+       |  }
+       |}
+       |""".stripMargin
+  )
+
+  check(
+    "insert companion object of trait inside parent class",
+    """|class Baz {
+       |  trait F<<>>oo {
+       |
+       |  }
+       |
+       |  class Bar {
+       |
+       |  }
+       |}
+       |""".stripMargin,
+    s"""|${CreateCompanionObjectCodeAction.companionObjectCreation}
+        |""".stripMargin,
+    """|class Baz {
+       |  trait Foo {
+       |
+       |  }
+       |
+       |  object Foo {
+       |
+       |  }
+       |
+       |  class Bar {
+       |
+       |  }
+       |}
+       |""".stripMargin
+  )
+
+  check(
+    "don't create existing companion object of trait",
+    """|trait Fo<<>>o{
+       |
+       |}
+       |
+       |object Foo {
+       |
+       |}
+       |""".stripMargin,
+    s"""|${ExtractRenameMember.title("trait", "Foo")}
+        |${CreateCompanionObjectCodeAction.companionObjectInfo}
+        |""".stripMargin,
+    """|trait Foo{
+       |
+       |}
+       |
+       |object Foo {
+       |
+       |}
+       |""".stripMargin,
+    selectedActionIndex = 1
+  )
+
+  check(
+    "don't create existing companion object of trait inside a parent object",
+    """|object Bar{
+       |  trait Fo<<>>o{
+       |
+       |  }
+       |
+       |  object Foo {
+       |
+       |  }
+       |
+       |  object Baz{
+       |
+       |  }
+       |
+       |}
+       |""".stripMargin,
+    s"""|${CreateCompanionObjectCodeAction.companionObjectInfo}
+        |""".stripMargin,
+    """object Bar{
+      |  trait Foo{
+      |
+      |  }
+      |
+      |  object Foo {
+      |
+      |  }
+      |
+      |  object Baz{
+      |
+      |  }
+      |
+      |}
+      |""".stripMargin
+  )
+
+  check(
+    "don't create existing companion object of trait inside a parent package",
```

Comment:
I wonder if we could reduce a bit the number of tests? For example we already test to not create a companion object if there is already one for classes, so it seems we are testing the same branches a number of times. 

I would test:
- create companion object for class
- create companion object for trait
- don't create companion object for class if it exists
- create companion object for a trait inside another object
- test creating companion object with Scala 3 and optional braces (this one needs to be done in Scala3CodeActionLspSuite since unit tests are able to only test the 2.13.8 version)

The reason here is that we always initialize the entire workspace in these tests, which makes our CI run a long while. If you want you could add a test suite that just tests the specific Code action (this is more akin to actual unit tests). I wanted to reduce our CI times for reworking the RenameSuite to do a similar thing.